UnitySetup/UnitySetup.psm1

```diff
@@ -2132,3 +2130,537 @@ function Get-UnityLicense {
         Write-Warning "Alias $($_.Name) already configured by $($alias.Source)"
     }
 }
+
+function Import-UPMConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [String]$ProjectManifestPath,
+        [int]$SearchDepth = 3
+    )
+
+    $ProjectManifestPaths = @()
+
+    if ((Get-Item $ProjectManifestPath) -is [System.IO.DirectoryInfo]) {
+        Write-Host "Path provided is not a manifest.json file ($ProjectManifestPath), will attempt search"
+        $FoundPaths = @(Get-Childitem -Path $ProjectManifestPath -Include manifest.json -File -Recurse -Depth $SearchDepth -ErrorAction SilentlyContinue)
+        foreach ($file in $FoundPaths) {
+            $ProjectManifestPaths += $file.FullName
+            if ($Verbose) { Write-Host "Found ($file.FullName)" }
+        }
+    }
+    else {
+        $ProjectManifestPaths += $ProjectManifestPath
+    }
+
+    if (([string]::IsNullOrEmpty($ProjectManifestPath)) -or (-not $(Test-Path $ProjectManifestPath))) {
+        Write-Error "Unable to find manifest.json file, please provide a path pointing directly to a Unity project's manifest.json or provide a path to a Unity project"
+        return @()
+    }
+
+    $scopedRegistrySet = [System.Collections.Generic.HashSet[string]]::new()
+    foreach ($ManifestPath in $ProjectManifestPaths) {
+        $manifest = Get-Content -Path $ManifestPath | ConvertFrom-Json
+
+        foreach ($scopedRegistry in $manifest.scopedRegistries) {
+            $url = $scopedRegistry.url -replace '/$', ''  
+            if ($url -like 'https://pkgs.dev.azure.com/*') {
+                $scopedRegistrySet.Add($url) | Out-Null
+            }
+        }
+    }
+
+    return [System.Collections.Generic.HashSet[string]]::new($scopedRegistrySet)
+}
+
+function Sync-UPMConfig {
+    [CmdletBinding()]
+    param(
+        [string[]]$scopedRegistryURLs,
+        [string[]]$tomlFilePaths,
```

Comment:
This cmdlet should take in a upm config loaded from the import cmdlet and validate all well known, e.g. urls have valid access tokens.